spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8SecretsPropertySource.java

```diff
@@ -64,6 +67,11 @@ public Fabric8SecretsPropertySource(KubernetesClient client, String name, String
 
 		}
 		catch (Exception e) {
+			if (failFast) {
```

Comment:
here is the concern I have and would like to hear your thoughts. Suppose we are reading `ConfigMapA`, at the same time we have enabled `fail-fast`, be that with or without retry, it is irrelevant to the point I am trying to make.

Now let's suppose that this `ConfigMapA` is read perfectly fine on the first attempt, no question here. At the same time, because of our flow, we will be trying to read `ConfigMapA-kubernetes` map. This happens because we add the `kubernetes` profile in the `AbstractKubernetesProfileEnvironmentPostProcessor::postProcessEnvironment` and in `KubernetesClientConfigMapPropertySource::getData` we generate and try to read from such a map (same idea applies for `fabric8`). 

Obviously such a ConfigMap is not present, in the vast majority of cases (if not all). We, on the other hand, will fail the application start-up, because of `fail-fast`. Users are not going to be happy and I guess even surprised.

There are two points here:

- there is a PR that I introduced that allows you to opt out of such `ConfigMapName-<ActiveProfiles>` readings, but it is not yet in `main` nor it will entirely fix this issue here. It allows to opt out of _all_ ConfigMap readings. So if you have a `ConfigMapB`, you can't say : "opt-out of `ConfigMapA-kubernetes`, but allow `ConfigMapB-kubernetes`"; neither do I think we want such granularity in that PR. 

- Initially, I thought that may be we should just ignore every failures related to `XXX-<ActiveProfile>`, but what if someone already relies on that? And this touches my first point above, of course. What if there are users that have `ConfigMapMyApp` and `ConfigMapMyApp-PROD`. When they were deploying to non PROD environments, the application would have started just fine (it only logged an Exception). If we merge this PR, and they enable retry (which is exactly what this PR wants them to do), it will now fail, because `ConfigMapMyApp-PROD` does not exist in non-prod environments. 

I will spent some more time trying to figure out an idea on how to approach this correctly (imho), but of course, your thoughts are absolutely a must too.

Reply:
Indeed, we don't fail when a ConfigMap or a Secret is not found. (Since Kubernetes Java clients do not throw exception on 404) Instead, we create an empty `PropertySource `as far as I see. So, our application still won't fail just because a ConfigMap or a Secret is missing even we enable `fail-fast`. Btw, it's obvious that retrying to load unexistent ConfigMap/Secret is pointless even we have retry enabled.

The question here is: "Should we fail on not found?" IMHO this question belongs to another topic. Actually, I considered adding a failure policy per config/secrets source like below:

```yaml
spring:
  cloud:
    kubernetes:
      config:
        fail-fast: true # when this is set to false, all failure-policies will be normalized to "never"
        sources:
         - name: some-config-map
           failure-policy: fail-on-not-found # fails when the config map is not present or any error occurs while loading it
         - name: another-config-map
           failure-policy: fail-on-error # (default) fail only when an error occurs while loading the config map. ignore if it's not found
         - name: optional-config-map
           failure-policy: never # ignore not found or any errors
```

But the same problem you mentioned above emerges with that design: When do we consider a config source as "not found"? Should we consider a config source as not found when any of the ConfigMaps with names `MyConfigMap`, `MyConfigMap-<profile>`, is not found? Or is it enough to say a config source "is found" if we are able to load any of these? I ended up leaving this behavior as-is.

IMHO we can leave resolving these questions to another feature since we don't actually break anything with this PR (if I'm not mistaken). What do you think?

Reply:
this is what I get when I comment without trying and solely based on my memory. I definitely used to see these errors on start-up, but I do not remember the context anymore :( I am really sorry for this comment without actually trying it first - I did not, for both clients and we are good indeed.  

Teşekkürler!

I am more than OK with this PR.

Reply:
No worries. And thank you very much for the review!